### PR TITLE
Preserve data

### PR DESCRIPTION
**Pull request by fatima-najafi**
On exit from app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly.
On startup of app load all data from files. handle the case when one or more of the files are missing.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -1,19 +1,12 @@
-require './person'
-require './book'
```
Comment (julie-ify) [inline app.rb:2]: - [ ] Please look into your app. I cannot create a person (student or teacher). Kindly fix that.

  image

- [ ] Given that I could not create a person I could not verify if the data is well saved when a person is created. Kindly ensure that works as listed in the instructions.

File: app.rb [hunk 2]
```diff
@@ -22,12 +15,31 @@ def list_books
...
+  end
+
+  def save_books
+    save_to_json('books.json', @books)
+  end
+
+  def load_books
+    load_from_json('books.json', @books)
+  end
+end
```
Comment (julie-ify) [inline app.rb]: - [ ] When a book is created this is what is saved in the book.json file. Kindly note that the data in the storage file should be human readable, displaying the book attributes, like the title, author and so on, not the memory object.

  image

File: app.rb [hunk 3]
```diff
@@ -1,19 +1,12 @@
...
+require_relative 'teacher'
+require_relative 'rental'
+require 'json'
 
-  # list of books
+# Module for managing books
+module BookManagement
   def list_books
     if @books.empty?
       puts 'No books found.'
```
Comment (julie-ify) [inline app.rb]: - [ ] When I create a book as seen above and I exit the app, when I try to list the books in the json file, the app crashes. Kindly fix this.

  image

- [ ] Kindly ensure your app works fine before re-submitting.

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @fatima-najafi and @AlyAtalla,

Good job so far!

## Highlights!
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mukhammadrizooff — APPROVED** [review]
Hi @TEAM, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

 

## Highlights
- No linter errors ✔️ 
- Correct GitFlow ✔️ 
- Good implementations ✔️
- Descriptive `README` file ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**